plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+   * {@inheritdoc}
+   */
+  public static function controllersInfo() {
+    return array(
+      '' => array(
+        // GET returns a list of entities.
+        \RestfulInterface::GET => 'index',
+        \RestfulInterface::HEAD => 'index',
+        // POST
+        \RestfulInterface::POST => 'setVariable',
+      ),
+      // We don't know what the ID looks like, assume that everything is the ID.
+      '^.*$' => array(
+        \RestfulInterface::GET => 'viewVariables',
+        \RestfulInterface::HEAD => 'viewVariables',
+        \RestfulInterface::PUT => 'setVariable',
+        \RestfulInterface::PATCH => 'setVariable',
+        \RestfulInterface::DELETE => 'remove',
+      ),
+    );
+  }
+
+  /**
+   * Defines default sort for variable names.
+   *
+   * By default, the array of variables returned by Drupal is already sorted
+   * in ascending order by variable name.
+   *
+   * @return array
+   *   Array keyed by the sort field, with the order ('ASC' or 'DESC') as value.
+   */
+  public function defaultSortInfo() {
+    return array();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getVariablesForList() {
+    global $conf;
+    $variables = $conf;
+
+    $this->applyListSort($variables);
+    $this->applyListPagination($variables);
+
+    return $variables;
+  }
+
+  /**
+   * Sort the list of variables by name.
+   *
+   * @param string $a
+   *   The first variable name for the comparison.
+   * @param string $b
+   *   The second variable name for the comparison.
+   */
+  public function nameSort($a, $b) {
+    return strcasecmp($a, $b);
+  }
+
+  /**
+   * Sort the list of variables by value.
+   *
+   * If you are comparing variables with the same value, note that the PHP
+   * sorting algorithm is not stable, and their order may be changed.
+   *
+   * @param mixed $a
+   *   The first data structure for the comparison.
+   * @param mixed $b
+   *   The second data structure for the comparison.
+   */
+  public function valueSort($a, $b) {
+    return;
+  }
+
+  /**
+   * Sort the list of variables.
+   *
+   * This data provider does not handle compound sorts; the last sort defined
+   * will be the one to take effect.
+   *
+   * @param array $variables
+   *   An array keyed by variable name, valued by unserialized variable value.
+   *
+   * @throws \RestfulBadRequestException
+   */
+  protected function applyListSort(array &$variables) {
+    $public_fields = $this->getPublicFields();
+
+    // Get the sorting options from the request object.
+    $sorts = $this->parseRequestForListSort();
+
+    $sorts = $sorts ? $sorts : $this->defaultSortInfo();
+
+    foreach ($sorts as $public_field_name => $direction) {
+      if (isset($public_fields[$public_field_name]['property'])) {
+        $property_name = $public_fields[$public_field_name]['property'];
+        if ($property_name == 'name') {
+          uksort($variables, array(__CLASS__, 'nameSort'));
+        }
+        elseif ($property_name == 'value') {
```

Comment:
Instead of an `if/else`, would it be possible to have a generic handle of this that would not need to separate by `$property_name`?

Maybe having a generic `$this->stringCompare` method?